File: frontend/src/api/videos.ts
import apiClient from "./client";
import type {
  VideoGenerationRequest,
  VideoGenerationResponse,
  VideoStatusResponse,
  JsonFileInfo,
  JsonFileStatusUpdate,
} from "@/types";

export const videoApi = {
  /**
   * 動画生成リクエストを送信
   */
  generate: async (
    data: VideoGenerationRequest
  ): Promise<VideoGenerationResponse> => {
    const response = await apiClient.post<VideoGenerationResponse>(
      "/videos/generate",
      data
    );
    return response.data;
  },

  /**
   * 動画生成のステータスを取得
   */
  getStatus: async (taskId: string): Promise<VideoStatusResponse> => {
    const response = await apiClient.get<VideoStatusResponse>(
      `/videos/status/${taskId}`
    );
    return response.data;
  },

  /**
   * ヘルスチェック
   */
  healthCheck: async (): Promise<{ status: string; service: string }> => {
    const response = await apiClient.get("/videos/health");
    return response.data;
  },

  /**
   * JSONファイル一覧を取得
   */
  listJsonFiles: async (): Promise<JsonFileInfo[]> => {
    const response = await apiClient.get<JsonFileInfo[]>(
      "/videos/json-files"
    );
    return response.data;
  },

  /**
   * JSONファイルの内容を取得
   */
  getJsonFile: async (filename: string): Promise<any> => {
    const response = await apiClient.get(`/videos/json-files/${encodeURIComponent(filename)}`);
    return response.data;
  },

  /**
   * JSONファイルのis_generatedステータスを更新
   */
  updateJsonFileStatus: async (
    filename: string,
    statusUpdate: JsonFileStatusUpdate
  ): Promise<JsonFileInfo> => {
    const response = await apiClient.patch<JsonFileInfo>(
      `/videos/json-files/${encodeURIComponent(filename)}/status`,
      statusUpdate
    );
    return response.data;
  },

  /**
   * JSONファイルを削除
   */
  deleteJsonFile: async (filename: string): Promise<{ success: boolean; message: string }> => {
    const response = await apiClient.delete<{ success: boolean; message: string }>(
      `/videos/json-files/${encodeURIComponent(filename)}`
    );
    return response.data;
  },
};

File: backend/app/api/videos/videos.py
"""動画生成API"""

from fastapi import APIRouter
from typing import List
from .videos_models import (
    VideoGenerationRequest,
    VideoGenerationResponse,
    VideoStatusResponse,
    JsonFileInfo,
    JsonFileStatusUpdate,
)
from .videos_handlers import (
    handle_generate_video,
    handle_get_video_status,
    handle_list_json_files,
    handle_get_json_file,
    handle_update_json_file_status,
    handle_delete_json_file,
)

router = APIRouter()


@router.post("/generate", response_model=VideoGenerationResponse)
async def generate_video(request: VideoGenerationRequest):
    """会話動画を生成する"""
    return await handle_generate_video(request)


@router.get("/status/{task_id}", response_model=VideoStatusResponse)
async def get_video_status(task_id: str):
    """動画生成のステータスを取得する"""
    return await handle_get_video_status(task_id)


@router.get("/health")
async def health_check():
    """動画生成APIのヘルスチェック"""
    return {"status": "healthy", "service": "videos"}


@router.get("/json-files", response_model=List[JsonFileInfo])
async def list_json_files():
    """outputs/json/ディレクトリ内のJSONファイル一覧を取得する"""
    return await handle_list_json_files()


@router.get("/json-files/{filename:path}")
async def get_json_file(filename: str):
    """指定されたJSONファイルの内容を取得する"""
    return await handle_get_json_file(filename)


@router.patch("/json-files/{filename:path}/status", response_model=JsonFileInfo)
async def update_json_file_status(
    filename: str, status_update: JsonFileStatusUpdate
):
    """JSONファイルのis_generatedステータスを更新する"""
    return await handle_update_json_file_status(filename, status_update)


@router.delete("/json-files/{filename:path}")
async def delete_json_file(filename: str):
    """JSONファイルを削除する"""
    return await handle_delete_json_file(filename)
